### ISSUE TITLE
[QA] Find VA Forms - Search MVP - Test Plan

### ISSUE THREAD

**Issue by tlei123** [department-of-veterans-affairs/va.gov-team#6900]
## Product/Feature summary

[VA Find VA Forms Search](https://www.va.gov/find-form) 
- Please see Mickin for creds

\[Provide brief summary of product/feature to be tested on Staging by VSA-QA. Also, be sure to include links to the Product/Feature Outline, Epic, and Pre-Launch Checklist.]

Product/Feature Outline: [GitHub doc](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/find-va-forms-redesign)
Product/Feature Epic: #1632
Product/Feature Design: \[GitHub folder](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/find-va-forms-redesign/design)
Product/Feature Launch Checklist: #5589

Test Plan (on TestRail TCMS): https://dsvavsp.testrail.io/index.php?/plans/view/105
\[Request VSA-Product-Team-specific TestRail account from @pjhill]

## Product-team SMEs
Team Lead: Jennifer Lee (@jenniferlee-dsva)
Acting PM: Micken Sahni (@MickinSahni)
Design/Research: Ryan Thurlwell (@rtwell)
Front-end: Nick Sullivan (@ncksllvn)
Back-end: \[N/A]
QA: Tze-chiu Lei (@tlei123)

## Proposed user-scenarios & Staging test-accounts

\[Provide a list of specific user scenarios/flows with matching Staging test-accounts (all the different interactive usage-flows and/or data-specific states) to be tested.  Or, link to existing issues/docs.]

1. **Testing colloquial search + PDF download** - Find and download the form that let's me apply as a Caregiver 
2. **Testing pagination + search** - Let me page through all forms related to "medical"
3. **Top tasks for frequently downloaded VA forms** - Complete top tasks available as online tools
4. **Manage your VA and other information online** - _Allow_ me to manage my personal VA information, but don't need to validate that I can do those tasks 

## Definition of Done

- [x] Staging test-accounts provided for all user scenarios.
- [x] Test Plan reviewed by Product Team.
- [x] Test Plan finalized per review feedback.
- [x] Actual Test Run(s) completed on Staging, and defects logged.

## How to configure this issue

- [x] Attached to a **Milestone** (when will this be completed?)
- [x] Attached to **Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [x] Labeled with **Team** (`vsa-authenticated-exp`, `vsa-benefits`, `vsa-benefits-2`, `vsa-caregiver`, `vsa-ebenefits`, `vsa-facilities`, `vsa-public-websites`, etc.)

**Comment by MickinSahni** [comment]
👋 @tzelei anything you need from us on this or will this be used to track your work? 
If there's nothing for a Public Websites person to do, I'd like to remove it from the sprint and remove the estimate.

**Comment by tlei123** [comment]
> 👋 @tzelei anything you need from us on this or will this be used to track your work?
> If there's nothing for a Public Websites person to do, I'd like to remove it from the sprint and remove the estimate.

@MickinSahni, yes please...  Can someone (PM or Designer) populate the **Proposed scenarios** list.

I've also remove the team label & estimates -- I need this to track my work and also conversations with SMEs.

Thanks much!

**Comment by MickinSahni** [comment]
Hey @tzelei, we have a new version of these designs coming (https://github.com/department-of-veterans-affairs/va.gov-team/issues/7097). 

I added testing scenarios above: https://github.com/department-of-veterans-affairs/va.gov-team/issues/6900#issue-582526494. I don't _think_ they'll change with the new designs, but you'll know best whether to move forward with this QA effort :)

**Comment by MickinSahni** [comment]
Moving to sprint 18

**Comment by tlei123** [comment]
Hi, @MickinSahni!  While I'm builidng test-cases for caregiver and medical scenarios, I'd like some clarifications please re. the 3rd and 4th scenarios:

1. Complete top tasks available as online tools: 
  - There's nothing saying "Top tasks" in the design screenshots in #6710 -- are we waiting on a design for this? Note: There is one search-result that shows an "Go to online tool" button, is that it?
  - What are "top tasks"?  The product outline only mentions them once, and doesn't go into any detail about them.

2. _Allow_ me to manage my personal VA information (sign-in not required):
  - Like top-tasks scenario above, there's nothing about this in the design screenshots -- is the design still forthcoming?
  - How does managing personal info without sign-in work?

Please let me know.  Thanks much!

**Comment by tlei123** [comment]
Hi, @rtwell!  Do you have a newer design that has the new "Top tasks" cards on the [Search MVP Beta](https://dev.va.gov/find-va-forms-beta/) page?

I checked your [PDFs](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/find-va-forms-redesign/design) and they still don't have those top task cards.

Thanks much!

**Comment by rtwell** [comment]
ah, I found the disconnect—my newer comps didn't include the bottom half. Apologies! I have updated the Sketch / desktop comps (I assume it's straightforward regarding the mobile but happy to add if helpful). PNG here for easy visual:

![image](https://user-images.githubusercontent.com/12564977/77654355-5ce39000-6f47-11ea-85af-dcd22062cac0.png)

**Comment by MickinSahni** [comment]
Hi Tze, in short:
1. Yes, it will Pass.
2. Just like 1., it will pass by validating that people can navigate to their expected destinations. This relates to the **Manage your VA and other information online** section near the bottom here: https://www.va.gov/find-form (Lmk if you're in need of credentials). The design is also available in Ryan's screenshot although the header doesn't match the header in the live version. 

Also, I updated [the initial comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6900#issue-582526494) to make the scenarios a bit more clear. 

> Hi, @MickinSahni! While I'm builidng test-cases for caregiver and medical scenarios, I'd like some clarifications please re. the 3rd and 4th scenarios:
> 
> 1. Complete top tasks available as online tools:
> 
> * For QA, please confirm that I can Pass this scenario after validating that their hyperlinks navigate to their expected destinations.
> 
> 1. _Allow_ me to manage my personal VA information (sign-in not required):
> 
> * There's nothing about this in the design screenshots or the beta page -- is the design still forthcoming?
> * How does managing personal info without sign-in work?  I'm confused about this.
> 
> Please let me know. Thanks much!

**Comment by tlei123** [comment]
@rtwell , re. the **Update basic information online** section, are the link destinations documented somewhere?

- Change you direct deposit information: <URL?>
- Change your address: <URL?>
- Request your medical record: <URL?>
- Request your military records, including DD214: <URL?>

Please let me know either where the documentation is (if it exists), or just fill in the URLs above.

Thanks much!

**Comment by rtwell** [comment]
@tlei123 that's a question for @jenniferlee-dsva 

my assumption those are the same links/top tasks as we have on va.gov home, but i'm not 100%

**Comment by tlei123** [comment]
> @tlei123 that's a question for @jenniferlee-dsva
> 
> my assumption those are the same links/top tasks as we have on va.gov home, but i'm not 100%

@jenniferlee-dsva , would you mind either confirming @rtwell 's assumption or providing the link URLs above (in my Comment just above his) please?

Thanks much!

**Comment by jenniferlee-dsva** [comment]
Hi @tlei123 - Yes, confirming @rtwell's comment. I'm uploading a doc here with the specific destination URLs for those and other links on this page. 

@MickinSahni and @ncksllvn - FYSA - this search page content was created sooooo long ago, that I see that there are a few copy changes that are needed prior to launch. (Sorry!) I'll create a ticket to get those implemented prior to launch, but maybe I should include it as part of Ryan's newer design updates that need to be implemented? Let me know - I'm happy to create a separate ticket or to add on to an existing ticket, whichever is easier for tracking and implementing.

**Comment by jenniferlee-dsva** [comment]
@tlei123 - @rtwell's PNG does not reflect most updated copy. I'm sorry - we created these so long ago and have since been making copy updates in drupal prod. Do you need a consolidated copydeck? I'm not sure if you are QA ing down to that level. LMK.

**Comment by jenniferlee-dsva** [comment]
I uploaded the updated copydeck with consolidated updates to the product repo. 
https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/find-va-forms-redesign/VA-forms-slp-conversion-paths-110519-updated-041020.docx  

(See highlighted text.) 

One thing: @ncksllvn I have in the older copydeck that the top task links (in the blue boxes) pointing to content pages, not the apply forms. Until we move the wizard to those apply pages, we'll need to continue pointing to the 'read more' content pages, sadly. The text is correct, but the URLs the 'read more's' are pointing to are not.

**Comment by MickinSahni** [comment]
👋 @tzelei does there need to be an a11y test case for [Find VA Forms Search](https://dsvavsp.testrail.io/index.php?/suites/view/9&group_by=cases:section_id&group_order=asc)?

We addressed [some "Must" items here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7096). Wondering if it's necessary. cc @stephorkandatx

**Comment by tlei123** [comment]
> 👋 @tzelei does there need to be an a11y test case for [Find VA Forms Search](https://dsvavsp.testrail.io/index.php?/suites/view/9&group_by=cases:section_id&group_order=asc)?

Yeah, I'll start creating the A11y cases. Will circle back and update here once they're ready.

**Comment by tlei123** [comment]
@jenstrickland / @ncksllvn , can you provide more specifics on the pagination-focus management that's called-out in #7096 please?

I'm confused by "Probably set focus back to the same spot."

Thanks much!

**Comment by jenstrickland** [comment]
@1Copenut would you please clarify for @tlei123 what is meant by "Probably set focus back to the same spot." — from feedback in https://github.com/department-of-veterans-affairs/va.gov-team/issues/6825 .

**Comment by 1Copenut** [comment]
@tlei123 I updated my comment on #6825 
> Focus will need to be managed when we have pagination too. Probably set focus back to the same spot ("Showing results for `SEARCH TERM`") when user presses a pagination link.

**Comment by tlei123** [comment]
@MickinSahni @lwalsh617 , please review the finalized [FF Test Cases](https://dsvavsp.testrail.io/index.php?/suites/view/9&group_by=cases:section_id&group_order=asc&group_id=71) at earliest convenience\*, and either approve \[click the **Test Plan reviewed by Product Team** checkbox in the main Description up top] or provide feedback.

Thanks much!

\*I'll most likely start test-runs here in a couple hours, betting on approval -- just need to have the QA reports ready for Staging Review on Wednesday.  I can always re-run the cases for which you may have feedback instead of approval.

**Comment by stephorkandatx** [comment]
@tlei123 As discussed, we have had some copy updates that will require updates to the test cases. The last version of the copy deck is with Jen's comment from yesterday, on ticket #7858 . 
I believe the primary test case that will need updating is for Personal Info.

**Comment by tlei123** [comment]
@stephorkandatx , test runs are now completed with updated test cases!  Yep, all the latest changes communicated above had been incorporated into the cases before runs started yesterday evening.

For you async Staging Review, here're the TestRail Test Plan and report links:
[Test Plan](https://dsvavsp.testrail.io/index.php?/plans/view/105)
[QA Coverage for References report](https://dsvavsp.testrail.io/index.php?/reports/view/60)
[Defects Summary report](https://dsvavsp.testrail.io/index.php?/reports/view/61)

1 styling defect logged -- #8200 -- assigned to you for planning/grooming.  :)

Thanks to you, @jenniferlee-dsva, @MickinSahni , @lwalsh617 for organizing/communicating all the latest changes!

cc @pjhill

**Comment by stephorkandatx** [comment]
Thanks @tlei123 !

**Comment by stephorkandatx** [comment]
@tlei123 with the defect logged, are we safe to close this ticket out?

**Comment by tlei123** [comment]
Yep. Closing now.
